functions/index.js

```diff
@@ -0,0 +1,65 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+
+admin.initializeApp(functions.config().firebase);
+const db = admin.firestore();
+
+const REQUEST_INVITE = 'REQUEST_INVITE';
+
+function createMemberFromOperation(operation) {
+    const {
+        applied,
+        block_at,
+        block_seq,
+        created_at,
+        creator_uid,
+        creator_mid,
+        data,
+        op_code,
+        op_seq
+    } = operation;
+
+    if (applied || op_code !== REQUEST_INVITE) {
+        return null;
+    }
+
+    const {
+        // unused to_uid,
+        // unused to_mid,
+        video_url,
+        full_name
+    } = data;
+
+    const newMember = db.collection('members').doc(creator_uid);
+    // By using `create`, we ensure that this operation will fail
```

Comment:
This also means also means that if they make some typo mistake during invite process then realize it once they get to their profile page (or later decide to request from someone else if original person does not confirm cause they are run over by bus), cannot instantly go back and change it. What if we instead allow people to update only if their `invite_confirmed` is still false? For consistency I think we would have to use `db.runTransaction`, first grab existing user (if it exists) then update (if safe to do so)

Reply:
After discussion - let's add functionality for someone to update their information from their profile page after requesting an invite - it seems like an odd flow for people to be able to request multiple invites from a single account.